test/jdk/java/nio/channels/DatagramChannel/StressNativeSignal.java

```diff
@@ -74,20 +70,37 @@ public void shutdown() {
         }
     }
 
+    public void waitForTestThreadsToStart() {
+
+        udpThread.waitTestThreadStart();
+        serverSocketThread.waitTestThreadStart();
+    }
+
     public class ServerSocketThread extends Thread {
         private volatile boolean shouldTerminate;
         private ServerSocket socket;
+        private volatile CountDownLatch threadStarted = new CountDownLatch(1);
 
         public void run() {
             try {
                 socket = new ServerSocket(1122);
+            } catch (Exception ignore) {
+                System.err.println("ServerSocketThread: caught exception " + ignore.getClass().getName());
+                System.err.println("continue ...");
```

Comment:
Continue if creating the ServerSocket fails looks bad. Can this move to the ServerSocketThread constructor so that it's not in the run method?

Reply:
I can do that, but the change is just maintaining the existing call flow, which would have any exception caught and continues with a stack trace dump. In this case we continue countdown the latch, an NPE is thrown, caught with stack trace and finish.
But we'll move to a Constructor as suggested.

Reply:
The catching of Exception looks like it's new code in this PR, did I mis-read it? In any case, I think it will be much better to do this in the constructor. Same thing in UDPThread where the code in the run method to catch IOException + continue is very surprising.

Reply:
yes, your point is well made , changes have been made as suggested ... I was just trying to say that I was following the idioms in the existing code ... the existing test code appears to be ambivalent to any exceptions, its sole purpose is to try to elicit a 316 error on macOS

Reply:
done